include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -151,8 +151,10 @@ double CMAES<SelectionPolicyType>::Optimize(
           pStep.slice(idx(j));
 
       // Calculate the objective function.
+      BoundaryTransform(pPosition.slice(idx(j)));
       pObjective(idx(j)) = selectionPolicy.Select(function, batchSize,
           pPosition.slice(idx(j)));
+      BoundaryTransformInverse(pPosition.slice(idx(j)));
```

Comment:
Hmm, I guess I'm a little confused here.  I would imagine it would be easier to take `pPosition.slice(idx(j))` and simply clamp it to be within the desired range, but I'm not sure what the reverse transformation is.  Can you clarify the functionality here a bit?  Maybe I misunderstood.